techs/css-sass.js

```diff
@@ -53,10 +54,17 @@ module.exports = require('enb/lib/build-flow').create()
                     successCb(cssResult);
                     deferred.resolve(cssResult);
                 } catch (err) {
-                    err = JSON.parse(err);
-                    err.formatted = util.format('%s in %s:%d:%d', err.message, err.file, err.line, err.column);
-                    errorCb(err);
-                    deferred.reject(err.formatted);
+                    if (typeof err === 'string') {
+                        err = JSON.parse(err);
+                    }
+                    var lines = sassSettings.data.split('\n');
+                    var ctx = lines.slice(err.line - 20, err.line).concat(
+                        '^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^',
+                        lines.slice(err.line + 1, err.line + 20)
```

Comment:
Maybe it's because of environment? I have error line :( well, I will try to find
out

Sent using CloudMagic [https://cloudmagic.com/k/d/mailapp?ct=pi&cv=6.4.27&pv=8.4]
On Thu, Aug 6, 2015 at 11:13 pm, enb-make/enb-sass
<reply@reply.github.com>
wrote:
In techs/css-sass.js
[https://github.com/enb-make/enb-sass/pull/5#discussion_r36459823] :

> @@ -53,10 +54,17 @@ module.exports = require('enb/lib/build-flow').create()
>                      successCb(cssResult);
>                      deferred.resolve(cssResult);
>                  } catch (err) {
> -                    err = JSON.parse(err);
> -                    err.formatted = util.format('%s in %s:%d:%d', err.message, err.file, err.line, err.column);
> -                    errorCb(err);
> -                    deferred.reject(err.formatted);
> +                    if (typeof err === 'string') {
> +                        err = JSON.parse(err);
> +                    }
> +                    var lines = sassSettings.data.split('\n');
> +                    var ctx = lines.slice(err.line - 20, err.line).concat(
> +                        '^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^',
> +                        lines.slice(err.line + 1, err.line + 20)


That should fix showing of line with error:

var ctx = lines.slice(err.line - 20, err.line).concat(                        
                        lines.slice(err.line, err.line + 1),
                        '^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^',
                         lines.slice(err.line + 1, err.line + 20)
                    ).join('\n');


—
Reply to this email directly or view it on GitHub [https://github.com/enb-make/enb-sass/pull/5/files#r36459823] .[https://github.com/notifications/beacon/AAtsbGBASLJhpV_spW-8kHU0-eJDhXpdks5ok7d3gaJpZM4FmsVO.gif]